patroni/dcs/etcd.py

```diff
@@ -264,8 +248,16 @@ def _get_machines_cache_from_srv(self, srv):
 
     def _get_machines_cache_from_dns(self, host, port):
         """One host might be resolved into multiple ip addresses. We will make list out of it"""
-        ret = ['{0}://{1}:{2}'.format(self.protocol, ip, port) for ip in set(self._dns_resolver.resolve(host))]
-        return list(set(ret)) if ret else ['{0}://{1}:{2}'.format(self.protocol, host, port)]
+        if self.protocol == 'http':
+            ret = []
+            for af, _, _, _, sa in self._dns_resolver.resolve(host, port):
+                host, port = sa[:2]
+                if af == socket.AF_INET6:
+                    host = '[{0}]'.format(host)
```

Comment:
underlying code performing http requests will do a second resolve anyway and therefore it should get everything it needs:

```python
In [1]: import socket

In [2]: socket.getaddrinfo('::1', 2379, 0, socket.SOCK_STREAM, socket.IPPROTO_TCP)
Out[2]: [(10, 1, 6, '', ('::1', 2379, 0, 0))]
```